### PR TITLE
Rfc30/cargo.toml update

### PR DESCRIPTION
**Pull request by thomas-k-cameron**
## Motivation and Context
part of RFC30

## Description
adds serde under aws_sdk_unstable feature gate

## Testing
no testing

## Checklist

- [x] I have updated `CHANGELOG.next.toml` if I made changes to the smithy-rs codegen or runtime crates
----

_By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice._

### CODE REVIEW

File: CHANGELOG.next.toml [hunk 1]
```diff
@@ -43,6 +43,15 @@ references = ["smithy-rs#1784", "smithy-rs#2074"]
 meta = { "breaking" = true, "tada" = true, "bug" = false }
 author = "rcoh"
 
+[[smithy-rs]]
+message = """Add serde crate to aws-smithy-types`.
```
Comment (jjant) [inline CHANGELOG.next.toml:47]:
```suggestion
message = """Add serde crate to `aws-smithy-types`.
```

File: CHANGELOG.next.toml [hunk 2]
```diff
@@ -43,6 +43,15 @@ references = ["smithy-rs#1784", "smithy-rs#2074"]
...
+
+It's behind feature gate `aws_sdk_unstable` which can only enabled via --cfg flag.
```
Comment (jjant) [inline CHANGELOG.next.toml:49]:
```suggestion
It's behind the feature gate `aws_sdk_unstable` which can only be enabled via a `--cfg` flag.
```

**Review by jjant — COMMENTED** [review]
Some small suggestions

**Comment by thomas-k-cameron** [comment]
@jjant 
Thank you for your suggestion!

**Review by Velfi — APPROVED** [review]

**Pull request merged**